Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp

```diff
@@ -102,6 +102,9 @@ class TIFFLoadingContext {
                         color = Color { TRY(decoded_strip->template read_value<u8>()), TRY(decoded_strip->template read_value<u8>()), TRY(decoded_strip->template read_value<u8>()) };
                     } else if (*m_metadata.samples_per_pixel() == 1) {
                         auto luminosity = TRY(decoded_strip->template read_value<u8>());
+
+                        if (m_metadata.photometric_interpretation() == PhotometricInterpretation::WhiteIsZero)
+                            luminosity = (1 << (*m_metadata.bits_per_sample())[0]) - 1 - luminosity;
```

Comment:
I get what you're doing here, but it's a bit weird. From what I can tell, you don't upsample the value you read (in the `upsample_to_8_bit()` sense in LibPDF's Renderer.cpp) so the value will be too dark, and you also always read one byte per pixel, while you should use that byte for several pixels if bits_per_sample is < 8 (…right?). And if you do that, this line should be just `luminosity = ~luminosity;` (?)

Reply:
Sure it'd work, but it's easier if you upsample before inverting since you can then just say `luminosity = ~luminosity;`. And if upsampling just isn't implemented yet, you can also say it now. Else support for bpp < 8 is kind of half-implemented, which is confusing.